### PR TITLE
BUG-111744 Audit events are missing for V3 api

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+    }
+
+    private boolean fillParams(String url, Map<String, String> params) {
+        if (checkAntiPattern(url)) {
+            return false;
+        }
+
+        Pattern pattern = Pattern.compile(getPattern());
+        Matcher matcher = pattern.matcher(url);
```
Comment (schfeca75) [inline core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java]: getPattern() should return with the Pattern itself, Pattern would be a member variable of the subclass.

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean checkAntiPattern(String url) {
+        if (getAntiPattern() != null) {
+            Pattern antiPattern = Pattern.compile(getAntiPattern());
+            Matcher antiMatcher = antiPattern.matcher(url);
```
Comment (schfeca75) [inline core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java]: getAntiPattern() should return with the Pattern itself, antiPattern would be a member variable of the subclass.

**Comment by schfeca75** [comment]
@biharitomi pls check the url patterns in the RestUrlParser subclasses...

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java [hunk 3]
```diff
@@ -190,45 +227,37 @@ private void sendStructuredEvent(RestRequestDetails restRequest, RestResponseDet
...
-        }
-        for (String urlRegex : urlPatterns) {
-            Pattern pattern = Pattern.compile(urlRegex);
-            Matcher matcher = pattern.matcher(url);
-            if (matcher.matches()) {
-                if (matcher.groupCount() >= 1) {
-                    params.put(RESOURCE_TYPE, matcher.group(1));
+        for (RestUrlParser restUrlParser : restUrlParsers) {
+            if (restUrlParser.fillParams(requestContext, params)) {
+                String workspaceId = params.get(WORKSPACE_ID);
```
[the remark this answers is not in the document]
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java [hunk 4]
```diff
@@ -190,45 +227,37 @@ private void sendStructuredEvent(RestRequestDetails restRequest, RestResponseDet
...
         return params;
     }
 
+    private void putResourceIdFromRepository(ContainerRequestContext requestContext, Map<String, String> params,
+            RestUrlParser restUrlParser, String workspaceId) {
+        for (Entry<String, WorkspaceResourceRepository<?, ?>> pathRepositoryEntry : pathRepositoryMap.entrySet()) {
+            String pathWithWorkspaceId = pathRepositoryEntry.getKey().replaceFirst("\\{.*\\}", workspaceId);
+            String requestUrl = restUrlParser.getUrl(requestContext);
+            if (('/' + requestUrl).startsWith(pathWithWorkspaceId)) {
+                WorkspaceResourceRepository<?, ?> resourceRepository = pathRepositoryEntry.getValue();
```
[the remark this answers is not in the document]
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/StructuredEventFilter.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java (hunk 1 above)
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java]: fixed

File: core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java (hunk 2 above)
Comment (sodre90) [reply core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java]: fixed

**Pull request merged**